pymongo/common.py

```diff
@@ -631,6 +711,37 @@ def get_validated_options(options, warn=True):
     return validated_options
 
 
+def _handle_option_deprecations(options):
+    """Issue appropriate warnings when deprecated options are present in the
+    options dictionary. Removes deprecated option key, value pairs if the
+    options dictionary is found to also have the renamed option."""
+    undeprecated_options = {}
+    for key, value in iteritems(options):
+        optname = str(key).lower()
+        if optname in URI_OPTIONS_DEPRECATION_MAP:
+            renamed_key = URI_OPTIONS_DEPRECATION_MAP[optname]
+            if any(renamed_key.lower() == n.lower() for n in options):
```

Comment:
You should make the list of lower cased options once outside of the for loop. Then this can be a simple membership test.